### PR TITLE
feat: add advanced settings section

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/1688

#### Changes proposed in this Pull Request

Adds the "Advanced settings" section to the settings page.

#### Testing instructions

- Ensure you have the "grouped settings" flag enabled in WCPay Dev Tools
- Go to your WCPay Home: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout
- Section is displayed

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/settings/advanced-settings/index.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+									<TextControl
+										label={ __(
+											'Custom font URL',
+											'woocommerce-payments'
+										) }
+										value={ customFontValue }
+										onChange={ setCustomFontValue }
+										autoComplete="off"
+										hideLabelFromVision
+									/>
```
Comment (Ferdev) [inline client/settings/advanced-settings/index.js:149]: What do you think about adding some format check for the font URL here?

File: client/settings/advanced-settings/index.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+							>
+								<TextareaControl
+									label={ __(
+										'Additional CSS styling',
+										'woocommerce-payments'
+									) }
+									value={ cssStylingValue }
+									onChange={ setCssStylingValue }
+									hideLabelFromVision
+								/>
```
Comment (Ferdev) [inline client/settings/advanced-settings/index.js:166]: What about trying to use `RichText` from https://github.com/Automattic/gutenberg/tree/trunk/packages/block-editor/src/components/rich-text? I'm not sure if we can just use it in a React component, but maybe we can use what that block is leveraging internally.

**Review by Ferdev — APPROVED** [review]
Looks good! Just a couple of comments, that we can maybe move to separate issues, but wanted to know your thoughts before creating them.

File: client/settings/advanced-settings/index.js (hunk 2 above)
Comment (frosso) [reply client/settings/advanced-settings/index.js:166]: That's a cool idea! How would the `RichText` component work in this scenario? From my understanding it is basically a WYSIWYG editor - is that accurate?

File: client/settings/advanced-settings/index.js (hunk 1 above)
Comment (frosso) [reply client/settings/advanced-settings/index.js:149]: That's a good idea! I modified the input's `type` to be `url`. I also added a placeholder text to help the user.
I tried adding a `pattern` property, but it doesn't look like the Gutenberg input allows that form of client-side validation.
I added an issue to our backlog to investigate that further 👍

File: client/settings/advanced-settings/index.js (hunk 2 above)
Comment (Ferdev) [reply client/settings/advanced-settings/index.js:166]: It's accurate! My idea was just to reuse most of what we have in the Code block in the Gutenberg editor. I think we would just need to import the `RichText` component from the `block-editor` package:

```
import { RichText } from '@wordpress/block-editor';
```
And then, something like this should do the trick:

```
<RichText
    tagName="code"
    multiline={ true }
    placeholder={ __(
        'Additional CSS styling',
        'woocommerce-payments'
    ) }
    value={ cssStylingValue }
    onChange={ setCssStylingValue }
    aria-label={ __( 'Code' ) }
    preserveWhiteSpace
/>
```

Of course, we would need to import the CSS styles for the component, and add the package `@wordpress/block-editor` to the `devDependencies`.

**Pull request merged**